tdigest_test.go

```diff
@@ -278,6 +290,62 @@ func TestTdigest_Reset(t *testing.T) {
 	}
 }
 
+func TestTdigest_OddInputs(t *testing.T) {
+	td := tdigest.New()
+	td.Add(math.NaN(), 1)
+	td.Add(1, math.NaN())
+	td.Add(1, 0)
+	td.Add(1, -1000)
+	if td.Count() != 0 {
+		t.Error("invalid value was alloed to be added")
+	}
+
+	// Infinite values are allowed.
+	td.Add(1, 1)
+	td.Add(2, 1)
+	td.Add(math.Inf(1), 1)
+	if q := td.Quantile(0.5); q != 2 {
+		t.Errorf("expected median value 2, got %f", q)
+	}
+	if q := td.Quantile(0.9); !math.IsInf(q, 1) {
+		t.Errorf("expected median value 2, got %f", q)
+	}
+}
+
+func TestTdigest_Merge(t *testing.T) {
+	td := tdigest.New()
+
+	// Repeat merges enough times to ensure we call compress()
+	numRepeats := 20
+	for i := 0; i < numRepeats; i++ {
+		for _, c := range NormalDigest.Centroids(nil) {
+			td.AddCentroid(c)
+		}
+		for _, c := range UniformDigest.Centroids(nil) {
+			td.AddCentroid(c)
+		}
+	}
+	addCentroids := td.Centroids(nil)
+
+	td.Reset()
+	for i := 0; i < numRepeats; i++ {
+		td.Merge(NormalDigest)
+		td.Merge(UniformDigest)
+	}
+
+	mergeCentroids := td.Centroids(nil)
+	if !reflect.DeepEqual(addCentroids, mergeCentroids) {
+		t.Error("Merge result differs from Add result")
+	}
+
+	// Empty merge does nothing
+	td.Merge(tdigest.New())
+	mergeCentroids = td.Centroids(nil)
+	if !reflect.DeepEqual(addCentroids, mergeCentroids) {
```

Comment:
Can you also update this test case to not compare tdigests be their centroids directly but by their responses to quantile queries?